runtime/parachains/src/hrmp.rs

```diff
@@ -447,8 +497,15 @@ pub mod pallet {
 		/// you to trigger the cleanup immediately for a specific parachain.
 		///
 		/// Origin must be Root.
-		#[pallet::weight(0)]
-		pub fn force_clean_hrmp(origin: OriginFor<T>, para: ParaId) -> DispatchResult {
+		///
+		/// Number of inbound and outbound channels for `para` must be provided as witness data.
+		#[pallet::weight(<T as Config>::WeightInfo::force_clean_hrmp(*_inbound, *_outbound))]
```

Comment:
the `*` is a deref. This is a bit detailed about how the macros expand. Unlike the body of the function where `_inbound` is `u32`, the weight expression is evaluated in a code similar to: 

```rust 
match call_enum {
	Call::force_clean_hrmp(para, _inbound, _outbound) => { /* weight expression */}
}
```

and this by default borrows all the enum arguments, even if they are `Copy`.

---

Validating this one is optional IMO, since the call is root. But a great catch, the non-root ones need to be validated.